src/components/renderers/layouts/masterdetail/masterdetail-directives.ts

```diff
@@ -47,128 +80,194 @@ class MasterDetailCollectionController {
     public get selectedChild() {
         return this.scope.selectedChild;
     }
-    public selectElement (child, value) {
-        this.scope.select(child, value.items);
+    public selectElement(child, schema): void {
+      if (child === undefined || schema === undefined) {
+        this.scope.select(undefined, undefined);
+      } else {
+        this.scope.select(child, schema.items);
+      }
+    }
+    public updateHasContents(scope): void {
+        if (scope.child === this.instance) {
+            scope.hasContents = true;
+            return;
+        }
+        let child = scope.child;
+        let schemaToCheck = scope.schema.items;
+        let keys = _.keys(this.getArraySubSchemas(schemaToCheck));
+        for (let key of keys) {
```

Comment:
I think this for-loop could be replaced with `scope.hasContents = _.some(keys, key => child[key] !== undefined ...`